app/testOnly/controllers/CustomLoginController.scala

```diff
@@ -67,17 +76,27 @@ class CustomLoginController @Inject()(implicit val appConfig: FrontendAppConfig,
                   "report-quarterly/income-and-expenses/view?origin=BTA"
                 val homePage = s"${appConfig.itvcFrontendEnvironment}/$redirectURL"
 
-                Redirect(homePage)
-                  .withSession(
-                    SessionBuilder.buildGGSession(AuthExchange(bearerToken = bearer,
-                      sessionAuthorityUri = auth)))
+                updateTestDataForOptOut(user.nino, CrystallisationStatus(appConfig)(postedUser.cyMinusOneCrystallisationStatus),
+                  ItsaStatusCyMinusOne(appConfig)(postedUser.cyMinusOneItsaStatus), postedUser.cyItsaStatus, postedUser.cyPlusOneItsaStatus).map {
+                  case _ =>
+                    Redirect(homePage)
+                      .withSession(
+                        SessionBuilder.buildGGSession(AuthExchange(bearerToken = bearer,
+                          sessionAuthorityUri = auth)))
+                }.recover {
+                  case ex =>
+                    val errorHandler = if (postedUser.isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+                    Logger("application")
+                      .error(s"[CustomLoginController][postLogin] - Unexpected response, status: - ${ex.getMessage} - ${ex.getCause} - ")
+                    errorHandler.showInternalServerError()
+                }
 
               case code =>
```

Comment:
Does this ever get called? I think everything will be cause by the first block? i.e. login only ever returns a tuple

Reply:
Yep good point, it does look that way. I think this would be outside of the scope of my ticket as I wouldn't know whether to remove it or fix it/what the original intent was